src/building/suggested.md

```diff
@@ -201,3 +201,74 @@ git worktree add -b my-feature ../rust2 master
 
 You can then use that rust2 folder as a separate workspace for modifying
 and building `rustc`!
+
+## Using nix-shell
+
+If you're using nix, you can use the following nix-shell to work on Rust:
+
+```nix
+{ pkgs ? import <nixpkgs> {} }:
+
+# This file contains a development shell for working on rustc.
+let
+  # Build configuration for rust-lang/rust. Based on `config.toml.example` from
+  # `1bd30ce2aac40c7698aa4a1b9520aa649ff2d1c5`.
+  config = pkgs.writeText "rustc-config" ''
+    profile = "tools"
```

Comment:
> Setting `profile = "tools"` is bad because not everyone wants that, but I don't know another good way to give a sample config.toml :/

I would set `profile = "compiler"` because this guide is primarily about rustc, and probably most readers are looking to contribute to the compiler itself. Then you can add a comment that says something like this:

> Other options are `library` for working on the standard library, `tools` for working on tools like rustdoc, [...].

Reply:
Good idea, done.